packages/editor/src/components/post-publish-panel/index.js

```diff
@@ -19,29 +19,10 @@ import PostPublishButton from '../post-publish-button';
 import PostPublishPanelPrepublish from './prepublish';
 import PostPublishPanelPostpublish from './postpublish';
 
-class PostPublishPanel extends Component {
+export class PostPublishPanel extends Component {
 	constructor() {
 		super( ...arguments );
 		this.onSubmit = this.onSubmit.bind( this );
-		this.state = {
-			loading: false,
-			submitted: false,
-		};
-	}
-
-	static getDerivedStateFromProps( props, state ) {
-		if (
-			state.submitted ||
-			props.isSaving ||
-			( ! props.isPublished && ! props.isScheduled )
```

Comment:
This logic wasn't accounting for the fact that an user can:

- schedule a post with a date in the future
- change its mind and modify the date to be in the past: after this, the user should be able to publish directly but wasn't.